dht/partition_filter.hh

```diff
@@ -28,11 +29,14 @@ public:
         // While token T is after a range Rn, advance the iterator.
         // iterator will be stopped at a range which either overlaps with T (if T belongs to node),
         // or at a range which is after T (if T doesn't belong to this node).
-        while (_it != _sorted_owned_ranges.end() && _it->after(t, dht::token_comparator())) {
-            _it++;
+        _it = std::lower_bound(_it, _sorted_owned_ranges.end(), t,
+                               [] (const dht::token_range& range, const dht::token& token) {
+            return range.after(token, dht::token_comparator());
+        });
+        if (_it == _sorted_owned_ranges.end()) {
+            return false;
         }
-
-        return _it != _sorted_owned_ranges.end() && _it->contains(t, dht::token_comparator());
+        return _it->contains(t, dht::token_comparator());
```

Comment:
nit: changing the return statement seems unjustified. Avoiding it would reduce the code churn and highlight exactly what changed.

Reply:
otherwise LGTM

Reply:
sure. but i think `_it->contains(t, dht::token_comparator())` deserves its own line. anyway, changed it back.